src/Symfony/Component/Notifier/Bridge/Twitter/TwitterTransport.php

```diff
@@ -0,0 +1,302 @@
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof TwitterOptions);
+    }
+
+    public function request(string $method, string $url, array $options): ResponseInterface
+    {
+        $url = 'https://'.str_replace('api.', str_starts_with($url, '/1.1/media/') ? 'upload.' : 'api.', $this->getEndpoint()).$url;
+
+        foreach (\is_array($options['body'] ?? null) ? $options['body'] : [] as $v) {
+            if (!$v instanceof DataPart) {
+                continue;
+            }
+
+            $formDataPart = new FormDataPart($options['body']);
+
+            foreach ($formDataPart->getPreparedHeaders()->all() as $header) {
+                $options['headers'][] = $header->toString();
+            }
+
+            $options['body'] = $formDataPart->bodyToIterable();
+
+            break;
+        }
+
+        $oauth = [
+            'oauth_consumer_key' => $this->apiKey,
+            'oauth_nonce' => self::$nonce = md5(self::$nonce ??= random_bytes(16)),
+            'oauth_signature_method' => 'HMAC-SHA1',
+            'oauth_timestamp' => time(),
+            'oauth_token' => $this->accessToken,
+            'oauth_version' => '1.0',
+        ];
+
+        $sign = $oauth + ($options['query'] ?? []) + (\is_array($options['body'] ?? null) ? $options['body'] : []);
+        ksort($sign);
+
+        $oauth['oauth_signature'] = base64_encode(hash_hmac(
+            'sha1',
+            implode('&', array_map('rawurlencode', [
+                $method,
+                $url,
+                implode('&', array_map(fn ($k) => rawurlencode($k).'='.rawurlencode($sign[$k]), array_keys($sign))),
+            ])),
+            rawurlencode($this->apiSecret).'&'.rawurlencode($this->accessSecret),
+            true
+        ));
+
+        $options['headers'][] = 'Authorization: OAuth '.implode(', ', array_map(fn ($k) => $k.'="'.rawurlencode($oauth[$k]).'"', array_keys($oauth)));
+
+        return $this->client->request($method, $url, $options);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $options['text'] = $message->getSubject();
+
+        try {
+            if (isset($options['attach'])) {
+                $options['media']['media_ids'] = $this->uploadMedia($options['attach']);
+                unset($options['attach']);
+            }
+
+            $response = $this->request('POST', '/2/tweets', ['json' => $options]);
+
+            $statusCode = $response->getStatusCode();
+        } catch (ExceptionInterface $e) {
+            throw new TransportException($e->getMessage(), $response, 0, $e);
```

Comment:
Is `$response` is defined here?

Reply:
if we want to guarantee that, the `$response = $this->request('POST', '/2/tweets', ['json' => $options]);` should be before the `try`. Otherwise, exceptions triggered during `->request()` will be caught without `$response` being defined

Reply:
good catch, fixed